### PR TITLE
Add "/entrypoint.sh" backwards-compat symlink in Oracle-based images

### LINKED ISSUE #884: Dockerfile difference between revision releases
The debian->oracle image switch omits the `/entrypoint.sh` symlink

The applicable line from a debian Dockerfile is the following:
https://github.com/docker-library/mysql/blob/master/5.7/Dockerfile.debian#L96

While that symlink is itself a backwards compat thing, I don't think its the sort of thing which should be removed from what is otherwise just a revision update from `5.7.37` to `5.7.38` (for tags like `:latest` I agree that all bets are off, but I think there should be a degree of consistency between releases in the same major.minor series)

If you were looking to keep the oracle Dockerfiles clean but still have those be the new foundation, perhaps until a new release series comes to exist, the former debian defaulted tags such as `:5.7` could build `FROM mysql:5.7-oracle`, simply adding things like the entrypoint symlink, a symlink for the config dir changes which were mentioned in other issues. It might also allow a more explicit means to warn of deprecated changes in that image series

### PR DESCRIPTION
**Pull request by tianon**
Fixes #884

Thanks for the report, @martynd! :heart:

### CODE REVIEW

File: template/Dockerfile.oracle [hunk 1]
```diff
@@ -130,6 +130,9 @@ RUN set -eux; \
 VOLUME /var/lib/mysql
 
 COPY docker-entrypoint.sh /usr/local/bin/
+{{ if [ "5.7", "8.0" ] | index(env.version) then ( -}}
+RUN ln -s usr/local/bin/docker-entrypoint.sh /entrypoint.sh # backwards compat
```
Comment (thaJeztah) [inline template/Dockerfile.oracle:134]: Should this use an absolute path? (just to be sure / explicit)?
```suggestion
RUN ln -s /usr/local/bin/docker-entrypoint.sh /entrypoint.sh # backwards compat
```

Comment (thaJeztah) [reply template/Dockerfile.oracle:134]: (I guess the image would unlikely get a WORKDIR changed/set, but if it would..)

Comment (tianon) [reply template/Dockerfile.oracle:134]: :shrug: it's put in the root, and this matches the Debian version :innocent:

Comment (tianon) [reply template/Dockerfile.oracle:134]: (Semi-intentional that it's not "pretty" so that we definitely get rid of it later :eyes:)

Comment (thaJeztah) [reply template/Dockerfile.oracle:134]: 🤷 alright, you win 😂💜

Comment (thaJeztah) [reply template/Dockerfile.oracle:134]: No hard(link) feelings 🤣

**Pull request merged**